packages/react/src/components/cookieConsent/portalModal/Portal.tsx

```diff
@@ -0,0 +1,63 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { CookieConsentContext } from '../CookieConsentContext';
+
+export function Portal({ rootId, children }) {
+  const cookieConsentContext = useContext(CookieConsentContext);
+  const hasOnlyRequiredConsents =
+    !cookieConsentContext.content.optionalCookies || cookieConsentContext.content.optionalCookies.groups.length === 0;
+  const shouldShowModal = !hasOnlyRequiredConsents && !cookieConsentContext.hasUserHandledAllConsents();
+
+  const containerElementRef = useRef<HTMLElement | null>(null);
+  const modalElementObserver = useRef<MutationObserver | null>(null);
+  const [isDomReady, setIsDomReady] = useState<boolean>(false);
+
+  const removePortalAndObservers = () => {
+    const containerElement = containerElementRef.current || document.getElementById(rootId);
+    containerElementRef.current = null;
+    if (containerElement && containerElement.childNodes.length === 0) {
+      containerElement.remove();
+    }
+    if (modalElementObserver.current) {
+      modalElementObserver.current.disconnect();
+      modalElementObserver.current = null;
+    }
+  };
+
+  const childNodeObserver = (mutationList: MutationRecord[]) => {
+    if (mutationList.filter((mutation) => mutation.removedNodes.length).length) {
+      removePortalAndObservers();
+    }
+  };
+
+  useEffect(() => {
+    if (!shouldShowModal) {
+      removePortalAndObservers();
+      return () => undefined;
+    }
+
+    let containerElement = document.getElementById(rootId);
+    if (!containerElement) {
+      containerElement = document.createElement('div');
+      containerElement.setAttribute('id', rootId);
+      document.body.appendChild(containerElement);
```

Comment:
The `appendChild` method appends the containerElement as the last child of body. We would wish that it would be the first child of the body? Because of accessibility.